### PR TITLE
Enable CI/CD for release branch

### PR DESCRIPTION
**Pull request by prmathur-microsoft**
This PR does the following 
* Adds a new key vault for fetching secrets for staging releases
* Identifies branch as 'release' and updates the ImageNamespace to "public"
* Enables nightly scheduled runs for 'release' branch

### CODE REVIEW

File: tools/e2etesting/steps/variables.yml [hunk 1]
```diff
@@ -5,4 +5,8 @@ variables:
   AgentPool: Azure-IoT-Manufacturing
   IAILocalFilename: Microsoft.Azure.IIoT.Deployment.exe
   TestPath: $(System.DefaultWorkingDirectory)\e2e-tests\IIoTPlatform-E2E-Tests
-  Runtime: 'win-x64'
\ No newline at end of file
+  Runtime: 'win-x64'
+  ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/') }}:
+    BranchName: $[ replace(variables['Build.SourceBranch'], 'refs/heads/', '') ]
+  ${{ if startsWith(variables['Build.SourceBranch'], 'refs/pull/') }}:
+    BranchName: $[ replace(variables['System.PullRequest.TargetBranch'], 'refs/heads/', '') ]
```
Comment (koepalex) [inline tools/e2etesting/steps/variables.yml:12]: Is this correct? checking SourceBranch to start with `/ref/pull` and then replace in target branch `refs/head/` with empty string without checking if traget branch startswith refs/head?

File: tools/e2etesting/steps/deployplatform.yml [hunk 2]
```diff
@@ -26,6 +26,15 @@ steps:
     SecretsFilter: 'ContainerRegistryPassword,ContainerRegistryServer,ContainerRegistryUsername'
     RunAsPreJob: true
 
+- task: AzureKeyVault@1
+  displayName: 'Load secrets from Release Key Vault as variables on Release Branch'
+  condition: eq(variables['BranchName'], 'release')
```
Comment (koepalex) [inline tools/e2etesting/steps/deployplatform.yml:31]: I would say this is not sufficient, e.g. when release branch is is hierachical (/release/Xyz), we need something like StartsWidth or in variables.yml some cut at the end.

File: tools/e2etesting/steps/variables.yml (hunk 1 above)
Comment (prmathur-microsoft) [reply tools/e2etesting/steps/variables.yml:12]: As per the discussion, System.PullRequest.TargetBranch wont be needed to identify branchname for triggering end to end tests.

File: tools/e2etesting/steps/deployplatform.yml (hunk 2 above)
Comment (prmathur-microsoft) [reply tools/e2etesting/steps/deployplatform.yml:31]: per discussion, we will need this for release/lts and also for older releases like release/2.7.206.

**Review by koepalex — APPROVED** [review]
:shipit:

**Pull request merged**